### ISSUE TITLE
Latest does not use icons from system theme?

### ISSUE THREAD

**Issue by thorsummoner** [mypaint/mypaint#541]
I was surprised to find the latest code (built on my system) doesn't use the system icon set specified, example:

![image](https://cloud.githubusercontent.com/assets/1320336/12047040/c7bdb934-ae79-11e5-8988-add0f695e703.png)

Notice the lower window, installed from ppa using system icons, whereas the latest build on top (with dark mode turned off) uses an incorrect set. (dark mode did not have an effect on the icon set used)

I would prefer programs use the system theme and icon set; this incorrect set per-vases the entire program and all its icons.

---

Could this be a configuration issue? I didn't see any settings to control this, or any notes in the runtime log about the inability to load the system theme.

---
#### log
```
INFO: mypaint: Installation layout: not installed, testing from within the source tree
INFO: mypaint: There is an old-style configuration area in u'/home/developer/.mypaint/'
INFO: mypaint: Its contents can be migrated to $XDG_CONFIG_HOME and $XDG_DATA_HOME if you wish.
INFO: mypaint: See the XDG Base Directory Specification for info.
INFO: lib.i18n: POSIX: LANG='en_US.UTF-8'
INFO: lib.i18n: POSIX: LANGUAGE=None
INFO: gui.brushmanager: Merging upstream brush changes into your collection.
INFO: gui.device: New device 'Razer Razer BlackWidow Ultimate 2013' (GDK_SOURCE_MOUSE, axes:3, class=gtk.gdk.X11DeviceXI2, vendor='?', product='?')
INFO: gui.device: New device 'Razer Razer BlackWidow Ultimate 2013' (GDK_SOURCE_MOUSE, axes:3, class=gtk.gdk.X11DeviceXI2, vendor='?', product='?')
INFO: gui.device: New device 'Razer Razer DeathAdder 2013' (GDK_SOURCE_MOUSE, axes:3, class=gtk.gdk.X11DeviceXI2, vendor='?', product='?')
INFO: gui.device: New device 'Razer Razer DeathAdder 2013' (GDK_SOURCE_MOUSE, axes:3, class=gtk.gdk.X11DeviceXI2, vendor='?', product='?')
INFO: gui.device: New device 'SynPS/2 Synaptics TouchPad' (GDK_SOURCE_TOUCHPAD, axes:4, class=gtk.gdk.X11DeviceXI2, vendor='?', product='?')
INFO: gui.device: New device 'Virtual core XTEST pointer' (GDK_SOURCE_MOUSE, axes:2, class=gtk.gdk.X11DeviceXI2, vendor='?', product='?')
INFO: gui.document: Initialized background from u'./backgrounds/10_soft_yellow.png'

(mypaint:27220): Gdk-WARNING **: gdk_window_set_icon_list: icons too large
WARNING: gui.keyboard: Ignoring keybinding for '<Actions>/BrushModifierActions/BlendModeMenu'
WARNING: gui.workspace: Empty tab group in workspace, not added
WARNING: gui.workspace: Empty tab group in workspace, not added
INFO: lib.document: Using u'load_ora' to load u'/home/developer/.mypaint/scratchpads/autosave.ora' (kwargs={'feedback_cb': <function gtk_main_tick at 0x7f7f43a8d0c8>})
INFO: lib.document: load_ora: u'/home/developer/.mypaint/scratchpads/autosave.ora'
ERROR: lib.layer.tree: Could not load any layer, document is empty.
INFO: lib.layer.tree: Adding an empty painting layer
INFO: lib.document: 0.117s load_ora total
INFO: gui.filehandling: Loaded scratchpad from u'/home/developer/.mypaint/scratchpads/autosave.ora'

(mypaint:27220): Gtk-CRITICAL **: _gtk_tree_view_column_autosize: assertion 'GTK_IS_TREE_VIEW (tree_view)' failed

(mypaint:27220): Gdk-WARNING **: gdk_window_set_icon_list: icons too large
WARNING: gui.freehand: Add motion event compression workaround: using evhack filter (<TiledDrawWidget object at 0x7f7f43a144b0 (TiledDrawWidget at 0x2c386b0)>, <gui.freehand.FreehandMode object at 0x7f7f38c1a4d0>)

```

**Comment by achadwick** [comment]
That's correct: the new icons are all symbolic, and have been for over a year. Check out the screenshots on the forums: <http://community.mypaint.org/t/screenshots-of-1-2-0-during-development> and in the old.

MyPaint's icons are themeable still, meaning theme developers can override them.

The ones you've highlighted are symbolic icons from the system theme, ironically enough. Perhaps they shouldn't be for consistency, and to reduce dependencies. Do you have an issue with those ones specifically?

**Comment by thorsummoner** [comment]
I take issue with all that can be provided by the system theme. I realize
there are many application specific ones, and many noon standard ones the
system may provide but may not be accessible in developer documentation,
ones not provided or too obscure are fine to provide defaults for.

IMO Gnome apps should use the system theme.

I'm both excited and threatened by all three new CSS providers and custom
extended bits of theming, and that stuff have a place, and are needed, but
please consider using the resources the person actually using the computer
asked for it at all possible.

I don't want to appear offended or upset. I was shocked to see a Gnome app,
which I just got so excited for, being python, gtk3 and amazing and all,
disregarding system icon theme as well as using a dark skin by default
without any system wide control as far as I could tell.

Basically I can't find things visually and effective when they don't look
like the icon I'm looking for.

**Comment by achadwick** [comment]
I was wrong earlier, sorry! Those three icons are indeed MyPaint-specific. _I do know my way around the codebase, honest! :blush:_

They're `mypaint-document-new-symbolic`, `mypaint-document-open-symbolic` and `mypaint-document-save-symbolic`, and they are derived from GNOME's symbolic icons set (I think, need to review the LICENSE file #546)

I understand what you're saying regarding familiarity. However I'm not going to change the icons so close to the release, after everyone has had over a year to get used to the new look and feed back on it.

> dark skin by default

MyPaint uses the dark variant of your chosen theme by default, if it has one. This can be overridden. It does not come with its own skin, and is not skinnable by itself. It respects your chosen UI theme, and your choice of variant.

The default dark variant is intentional. The UI should recede away from the user. If there is cool/eerie monitor glow happening, it should be from your artwork and not the UI surrounds ☺  The idea is similar to video applications, which try to encourage similar immersion. I note that GNOME's Videos app does this.

Evidently it's common practice to use darker themes and simpler icons for art apps too these days (links redirect to Google image search). We're not alone.

* https://duckduckgo.com/?q=%21i+photoshop+cs6
* https://duckduckgo.com/?q=%21i+blender+ui

There is a user-facing toggle in the preferences where you can tell MyPaint to use the light variant instead, if you prefer. However MyPaint adopts the dark variant for its default, just like a GNOME video player application.

MyPaint isn't strictly a GNOME app: it runs on Windows, OS X, Xfce4, younameit.

**Comment by thorsummoner** [comment]
Thanks for the info, I really need to learn how to work with Linux icon themes. If you have any leads on resources to learn how to build them I would be plenty happy writing my own theme overrides into the system provided my paying specific overrides

**Comment by achadwick** [comment]
Sorry, I'm just a Python guy with a bit of C++/C geekiness. One of us might be able to answer theming questions or share resources on the forums: <http://community.mypaint.org/>.
